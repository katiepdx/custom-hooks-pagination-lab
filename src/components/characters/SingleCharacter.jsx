// single character - name and image - snapshot test 

import React from 'react'
import propTypes from 'prop-types'

const SingleCharacter = ({ image, name }) => {
  // return a div for a single character
  return (
    <div>
      <h3>You choose {name}!</h3>
      <img src={image} alt={name} />
      <p>This is {name}</p>
    </div>
  )
}

SingleCharacter.propTypes = {
  image: propTypes.string.isRequired,
  name: propTypes.string.isRequired
}

export default SingleCharacter
