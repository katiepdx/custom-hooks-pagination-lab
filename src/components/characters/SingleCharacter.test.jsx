// tests that a singleCharacter item renders to the screen using test data

import React from 'react'
import { render } from '@testing-library/react'
import SingleCharacter from './SingleCharacter'

describe('Tests SingleCharacter component', () => {
  it('should render single character item to the page', () => {
    const { asFragment } = render(<SingleCharacter
      image="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      name="Rick Sanchez"
    />)

    expect(asFragment()).toMatchSnapshot();
  })
})
