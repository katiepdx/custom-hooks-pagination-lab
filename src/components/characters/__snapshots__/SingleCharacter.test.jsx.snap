// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests SingleCharacter component should render single character item to the page 1`] = `
<DocumentFragment>
  <div>
    <img
      alt="Rick Sanchez"
      src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
    />
    <p>
      This is Rick Sanchez
    </p>
  </div>
</DocumentFragment>
`;
